.platform/hooks/predeploy/06_fetch_ssm_parameters.sh

```diff
@@ -0,0 +1,116 @@
+#!/bin/bash
+
+# Exit on first error
+set -e
+
+# Make sure the local directory exists
+mkdir -p /tmp/isomer
+
+# If the temp .isomer.env file exists, remove it
+if [ -f "/tmp/isomer/.isomer.env" ]; then
+    rm /tmp/isomer/.isomer.env
+fi
+
+# Create EFS directory if it does not exist
+if [ ! -d "/efs/isomer" ]; then
+    mkdir -p /efs/isomer
+    chown webapp:webapp /efs/isomer
+fi
+
+ENV_TYPE=$(/opt/elasticbeanstalk/bin/get-config environment -k SSM_PREFIX)
+echo "ENV TYPE: $ENV_TYPE" >> /tmp/ssm-type.txt
+
+# List of all env vars to fetch
+ENV_VARS=(
+  "AUTH_TOKEN_EXPIRY_DURATION_IN_MILLISECONDS"
+  "AWS_BACKEND_EB_ENV_NAME"
+  "AWS_REGION"
+  "CLIENT_ID"
+  "CLIENT_SECRET"
+  "CLOUDMERSIVE_API_KEY"
+  "COOKIE_DOMAIN"
+  "DB_ACQUIRE"
+  "DB_MAX_POOL"
+  "DB_MIN_POOL"
+  "DB_TIMEOUT"
+  "DB_URI"
+  "DD_AGENT_MAJOR_VERSION"
+  "DD_ENV"
+  "DD_LOGS_INJECTION"
+  "DD_SERVICE"
+  "DD_TAGS"
+  "DD_TRACE_STARTUP_LOGS"
+  "E2E_TEST_GH_TOKEN"
+  "E2E_TEST_REPO"
+  "E2E_TEST_SECRET"
+  "EFS_VOL_PATH"
+  "ENCRYPTION_SECRET"
+  "FF_DEPRECATE_SITE_QUEUES"
+  "FRONTEND_URL"
+  "GGS_EXPERIMENTAL_TRACKING_SITES"
+  "GITHUB_BUILD_ORG_NAME"
+  "GITHUB_BUILD_REPO_NAME"
+  "GITHUB_ORG_NAME"
+  "GROWTHBOOK_CLIENT_KEY"
+  "INCOMING_QUEUE_URL"
+  "ISOMERPAGES_REPO_PAGE_COUNT"
+  "JWT_SECRET"
+  "MAX_NUM_OTP_ATTEMPTS"
+  "MOCK_AMPLIFY_DOMAIN_ASSOCIATION_CALLS"
+  "MUTEX_TABLE_NAME"
+  "NETLIFY_ACCESS_TOKEN"
+  "NODE_ENV"
+  "OTP_EXPIRY"
+  "OTP_SECRET"
+  "OUTGOING_QUEUE_URL"
+  "POSTMAN_API_KEY"
+  "POSTMAN_SMS_CRED_NAME"
+  "REDIRECT_URI"
+  "SESSION_SECRET"
+  "SGID_CLIENT_ID" 
+  "SGID_CLIENT_SECRET" 
+  "SGID_PRIVATE_KEY" 
+  "SGID_REDIRECT_URI"
+  "SITE_CREATE_FORM_KEY"
+  "SITE_LAUNCH_DYNAMO_DB_TABLE_NAME"
+  "SITE_LAUNCH_FORM_KEY"
+  "SITE_PASSWORD_SECRET_KEY"
+  "SSM_PREFIX"
+  "STEP_FUNCTIONS_ARN"
+  "SYSTEM_GITHUB_TOKEN"
+  "TEST_VAR"
+)
+
+echo "Set AWS region"
+aws configure set default.region ap-southeast-1
+
+for ENV_VAR in "${ENV_VARS[@]}"; do
+  echo "Fetching ${ENV_VAR} from SSM"
+  VALUE=$(aws ssm get-parameter --name "${ENV_TYPE}_${ENV_VAR}" --with-decryption --query "Parameter.Value" --output text)
+  echo "${ENV_VAR}=${VALUE}" >> /tmp/isomer/.isomer.env
+  echo "Saved ${ENV_VAR}"
+done
+
+# Use flock to ensure that the EFS file is locked during the copy operation
+(
+  flock -n 200 || exit 1
+
+  # Copy the local file to EFS
+  echo "Copying local env file to EFS"
+  cp /tmp/isomer/.isomer.env /efs/isomer/.isomer.env
+
+  # Ensure the file on EFS is owned by webapp so it has access
+  chown webapp:webapp /efs/isomer/.isomer.env
+
+) 200>/efs/isomer/.isomer.lock
```

Comment:
what does the number indicate here? the [man page](https://man7.org/linux/man-pages/man1/flock.1.html) doesn't seem to have it

Reply:
200 is just a file descriptor - can be an arbitrary value that is associated with a file

Reply:
updated PR desc

Reply:
ps, how do we ensure that the fd is always given to flock?

Reply:
Most commonly used file descriptors are:

0: Standard input (stdin)
1: Standard output (stdout)
2: Standard error (stderr)

Choosing a high number minimises chances of conflict. There is still a super low probability it is unavailable for some reason, but for purpose of deploy script such as this I think we may not need to over-engineer

Reply:
I can add a check to find a available file descriptor using a loop or smth, but yea, let me know if you think its necessary

Reply:
Highly unlikely - tried listing open FDs using `ls -l /proc/$$/fd`

it only has 4 entries, of which 0, 1, 2 and 255 are used

Reply:
@seaerchin to edit my explanation - File descriptors are local to each process. Two different processes can use the same file descriptor number simultaneously without interfering with each other